### PR TITLE
Care for https://me.sap.com/notes/3450531 Table X030L problem

### PR DESCRIPTION
**Pull request by DerGuteWolf**
get_ddic_header( ) is released and thus no syntax error but throws a runtime error, cf. https://me.sap.com/notes/3450531 .
Need to use steampunk way in steampunk and conventional way in non-steampunk.

### CODE REVIEW

**Comment by sbcgua** [comment]
Looks good except the DTEL. Would it be possible to avoid creating a new DDIC type? Especially since it is used in test only?

**Comment by DerGuteWolf** [comment]
I know, but for testing if a timestamp ddic type works I need a ddic type. I could try dynamically creating one and deleting it afterwards, but would be very cumbersome..

**Comment by larshp** [comment]
try https://abapedia.org/steampunk-2305-api-intersect-702/timestampl.dtel.html or https://abapedia.org/steampunk-2305-api-intersect-702/timestamp.dtel.html, both exist in 702 and is released in steampunk

also more examples welcome in https://github.com/heliconialabs/steampunkification

**Comment by DerGuteWolf** [comment]
Hi @larshp,

thanks for the hint and I know those two, but they are already dealt with in the respective first comparison. I need a different named dtel which makes the second check for domain name necessary. 

Unfortunately this is a real mess, cf eg https://abapedia.org/steampunk-2305-api-intersect-702/tzntstmpl.dtel.html the Doma link gives 403 because the Doma is not in the intersect. Perhaps you could check this and instead of the link generate a unreleased hint. 

Regards,
Wolfgang

**Comment by sbcgua** [comment]
Adding a DDIC object immediately makes this package heavier. I think it is even better to skip this particular UT than adding a UT-specific and the only DDIC object.

Or targeting the UT to 702 - so create variable dynamically with some standard named type, catching the exception in the steampunk env and treating it as tolerable.

**Comment by DerGuteWolf** [comment]
@sbcgua The problem is that I need this at least to be syntax error free in steampunk - use of unreleased stuff is a syntax error... Also as the steampunk is a bit complicated I would definitly to have this covered by the test.

**Comment by DerGuteWolf** [comment]
I will try once more to find a suitable alternative however

**Comment by DerGuteWolf** [comment]
image
Domain TZNTSTMPL h has a few more releases uses in steampunk, however none of these are available in 750

**Comment by DerGuteWolf** [comment]
I have an idea: I will dynamically create a variable, steampunk way with steampunk released DTEL and non-steampunk way with DTEL available in at least in 750. Will however take some time.... let's leave this pull request open until then

**Comment by DerGuteWolf** [comment]
Thanks for your input!

Now without ddic and with dynamic unit test. I am not sure if the unit test could be simplified a bit more, but it works now in 750 and steampunk. I also added performance improvement for dtel TZNTSTMPS/TZNTSTMPL and an appropriate unit test.

File: src/zcl_mustache_render.clas.testclasses.abap [hunk 1]
```diff
@@ -178,6 +187,30 @@ class ltcl_mustache_render implementation.
         cl_abap_unit_assert=>fail( lx->msg ).
     endtry.
 
+    try.
+        data(lv_elem) = cl_abap_datadescr=>describe_by_name( 'BALTIMSTMP' ).
```
Comment (sbcgua) [inline src/zcl_mustache_render.clas.testclasses.abap]: 740+

File: src/zcl_mustache_render.clas.testclasses.abap [hunk 2]
```diff
@@ -178,6 +187,30 @@ class ltcl_mustache_render implementation.
...
+      catch CX_SY_RTTI_NO_CHECK.
+        lv_elem = cl_abap_datadescr=>describe_by_name( 'RAP_CP_ODATA_V2_EDM_DATETIME' ).
+    endtry.
+    try.
+        DATA(lt_comp) = VALUE cl_abap_structdescr=>component_table( ( name = 'TIMESTAMP_DDIC' type = CAST #( lv_elem ) ) ).
```
Comment (sbcgua) [inline src/zcl_mustache_render.clas.testclasses.abap]: 740+

**Review by sbcgua — CHANGES_REQUESTED** [review]
Looks good now. Just downport the syntax to 702 and it's good to go. Thank you !

**Comment by DerGuteWolf** [comment]
Ok, will do that. I have no 702 system currently. Wasn't there some possibility to check for older syntax in a newer system?

**Review by sbcgua — APPROVED** [review]
Works well on 731. :+1:

Thanks again ! Merging ...

**Pull request merged**